ClangSharpPInvokeGenerator/CursorWriter.cs

```diff
@@ -41,1727 +34,1262 @@ public void Dispose()
             }
 
             _outputBuilders.Clear();
-
-            Debug.Assert(_attachedData.Count == 0);
-            Debug.Assert(_processingCursors.Count == 0);
-            Debug.Assert(_predicatedCursors.Count == 0);
-
             _visitedCursors.Clear();
         }
 
-        protected override bool BeginHandle(CXCursor cursor, CXCursor parent)
+        public void Visit(TranslationUnit translationUnit)
         {
-            if (_predicate(cursor))
-            {
-                _predicatedCursors.Push(cursor);
-            }
-            else if ((_predicatedCursors.Count == 0) && (cursor.Kind != CXCursorKind.CXCursor_UnexposedDecl))
-            {
-                return false;
-            }
-
-            _processingCursors.Push(cursor);
-            _visitedCursors.Add(cursor);
+            Debug.Assert(_outputBuilder is null);
+            _visitedCursors.Add(translationUnit);
+            VisitChildren(translationUnit);
+        }
 
-            switch (cursor.Kind)
+        private string EscapeName(string name)
```

Comment:
The helper functions for generating names are the same as before, but pulling from the `Cursor` types rather than directly from `CXCursor`.